### PR TITLE
Improve token error reporting in Prosody JWT plugin

### PR DESCRIPTION
**Pull request by paweldomas**
Adjusts log messages and informs about missing 'room' claim.

### CODE REVIEW

File: prosody-plugins/mod_auth_token.lua [hunk 1]
```diff
@@ -80,6 +80,9 @@ function provider.get_sasl_handler(session)
 		-- here we check if 'room' claim exists
 		local room, roomErr = token_util.get_room_name(token, appSecret);
 		if room == nil then
+            if roomErr == nil then
```
Comment (emcho) [inline prosody-plugins/mod_auth_token.lua]: why exactly do we need the second if statement here?

File: prosody-plugins/token/util.lib.lua [hunk 2]
```diff
@@ -23,15 +23,15 @@ local function _verify_token(token, appId, appSecret, roomName)
 
 	local issClaim = claims["iss"];
 	if issClaim == nil then
-		return nil, "Issuer field is missing";
+		return nil, "'iss' claim is missing";
```
Comment (emcho) [inline prosody-plugins/token/util.lib.lua]: I would change this to:
The Issuer (iss) part of the claim is missing

File: prosody-plugins/token/util.lib.lua [hunk 3]
```diff
@@ -23,15 +23,15 @@ local function _verify_token(token, appId, appSecret, roomName)
...
 	end
 	if issClaim ~= appId then
 		return nil, "Invalid application ID('iss' claim)";
 	end
 
 	local roomClaim = claims["room"];
 	if roomClaim == nil then
-		return nil, "Room field is missing";
+		return nil, "'room' claim is missing";
```
Comment (emcho) [inline prosody-plugins/token/util.lib.lua]: I would change this to:
The Room (room) part of the claim is missing.

File: prosody-plugins/mod_auth_token.lua (hunk 1 above)
Comment (paweldomas) [inline prosody-plugins/mod_auth_token.lua]: When we reach this point, the room is nil, so we check if there was any error during token verification. If the error is nil it means that the claim is missing.

File: prosody-plugins/token/util.lib.lua (hunk 2 above)
Comment (paweldomas) [inline prosody-plugins/token/util.lib.lua]: Claim is the particular entry/field like 'room', 'iss', so writing "part of the claim" is not accurate

**Pull request merged**